### ISSUE TITLE
Confidence Loss

### ISSUE THREAD

**Issue by GitTsao** [experiencor/basic-yolo-keras#88]
Something seems off with the confidence loss. I've been having trouble running your Blood Cell detection notebook. The model seems to always converge not predicting any boxes as the best answer. I've tried scaling up the object scale really high to try and emphasize that boxes need to be predicting, but I'm not having much luck

**Comment by GitTsao** [comment]
Also new to git issues and where to put things, but if you'd like to make your loss work for non-square images. I believe this can fix your loss function.

cell_x = tf.to_float(tf.reshape(tf.tile(tf.range(GRID_W), [GRID_H]), (1, GRID_H, GRID_W, 1, 1)))
cell_y = tf.to_float(tf.transpose(tf.reshape(tf.tile(tf.range(GRID_H), [GRID_W]),
                                                 (1, GRID_W, GRID_H, 1, 1)), (0, 2, 1, 3, 4)))

Instead of this,
cell_x = tf.to_float(tf.reshape(tf.tile(tf.range(GRID_W), [GRID_H]), (1, GRID_H, GRID_W, 1, 1)))
cell_y = tf.transpose(cell_x, (0,2,1,3,4))

Figure I'd contribute back to the code since, I'm asking questions :P

**Comment by experiencor** [comment]
@GitTsao So it changes the code and it stops working? Or it's not even working with the original square version?

**Comment by GitTsao** [comment]
Sorry for the delay, I was doing more investigation. I did get networks to converge now, by altering the SCALE parameters at the top and removing the divisors from the loss function. My data was pretty heavy on non-targets so I had to make some alterations. 

Curious though, did you deviate from the original loss function because of experiments you ran?

**Comment by experiencor** [comment]
@GitTsao From my experiments, I find that that the network is more stable with the divisors. Training with no divisors and Adam usually leads to NaN loss.
